native/cocos/renderer/gfx-vulkan/VKGPUObjects.h

```diff
@@ -148,10 +147,19 @@ struct CCVKGPUTexture : public CCVKGPUDeviceObject {
     uint32_t size = 0U;
     uint32_t arrayLayers = 1U;
     uint32_t mipLevels = 1U;
-    SampleCount samples = SampleCount::ONE;
+    SampleCount samples = SampleCount::X1;
     TextureFlags flags = TextureFlagBit::NONE;
     VkImageAspectFlags aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
-    bool memoryless = false;
+
+    /*
+     * allocate and bind memory by Texture.
+     * If any of the following conditions are met, then the statement is false
+     * 1. Texture is a swapchain image.
+     * 2. Texture has flag LAZILY_ALLOCATED.
+     * 3. Memory bound manually bound.
+     * 4. Sparse Image.
+     */
+    bool allocateMemory = true;
```

Comment:
It is not a good variable name. It is more like a function name. May be change to `needToAllocateMemory` is better.

Reply:
named to `memoryAllocated`